keras_nlp/benchmarks/text_generation.py

```diff
@@ -0,0 +1,215 @@
+        vocabulary_size=vocab_size,
+        sequence_length=max_length,
+        embedding_dim=embed_dim,
+        mask_zero=True,
+    )(inputs)
+    # Transformer decoders.
+    for _ in range(num_layers):
+        x = keras_nlp.layers.TransformerDecoder(
+            num_heads=num_heads,
+            intermediate_dim=ff_dim,
+        )(x)
+    # Output.
+    outputs = keras.layers.Dense(vocab_size)(x)
+    model = keras.Model(inputs=inputs, outputs=outputs)
+    return model
+
+
+def generate_text(
+    decoding_fn,
+    token_probability_fn,
+    prompt,
+    text_gen_args,
+    jit_compile,
+):
+    class TestModel(tf.keras.Model):
+        def call(self, inputs):
+            generated = decoding_fn(
+                token_probability_fn=token_probability_fn,
+                prompt=inputs,
+                **text_gen_args,
+            )
+            return generated
+
+    test_model = TestModel()
+    test_model.compile(jit_compile=jit_compile)
+
+    t0 = time.time()
+    _ = test_model.predict(prompt)
+    return time.time() - t0
+
+
+def main():
+    keras.utils.set_random_seed(SEED)
+
+    ds = generate_random_ds(
+        vocab_size=DATASET_ARGS["vocab_size"],
+        num_samples=DATASET_ARGS["num_samples"],
+        batch_size=DATASET_ARGS["batch_size"],
+        seed=SEED,
+    )
+
+    model = build_model(
+        vocab_size=DATASET_ARGS["vocab_size"],
+        max_length=MODEL_ARGS["max_length"],
+        embed_dim=MODEL_ARGS["embed_dim"],
+        num_layers=MODEL_ARGS["num_layers"],
+        num_heads=MODEL_ARGS["num_heads"],
+        ff_dim=MODEL_ARGS["ff_dim"],
+    )
+
+    def token_logits_fn(inputs):
+        output = model(inputs)
+        return output[:, -1, :]  # return next token logits
+
+    print("*************************************\n")
+
+    with open("./results.csv", "w") as res_handler:
+        res_handler.write("decoding_strategy,execution_method,time\n")
+        for test_run in TEST_RUNS:
+            decoding_fn = test_run["decoding_fn"]
+            decoding_strategy = decoding_fn.__name__
+
+            for execution_method in test_run["execution_methods"]:
+                print(f"Running {decoding_strategy} in {execution_method} mode")
+
+                if execution_method == "graph":
+                    jit_compile = False
+                elif execution_method == "xla":
+                    jit_compile = True
+
+                time_taken = generate_text(
+                    decoding_fn=decoding_fn,
+                    token_probability_fn=token_logits_fn,
+                    prompt=ds,
+                    text_gen_args=test_run["args"],
+                    jit_compile=jit_compile,
+                )
+                print("Time taken: ", time_taken)
+                res_handler.write(
+                    f"{decoding_strategy},{execution_method}," f"{time_taken}\n"
+                )
+                print()
+            print("*************************************")
+
```

Comment:
Since you are writing results to a file, let's add logging at the end: 
```
writing benchmark results to {path}
```
so users are notified that a new file is generated.